cirq-core/cirq/ops/raw_types.py

```diff
@@ -211,11 +211,13 @@ def validate_args(self, qubits: Sequence['cirq.Qid']) -> None:
         """
         _validate_qid_shape(self, qubits)
 
-    def on(self, *qubits: Qid) -> 'Operation':
+    def on(self, *qubits: Qid, **kwargs) -> 'Operation':
```

Comment:
This is a breaking change because all user defined classes which override the `on` method as `def on(self, *qubits: Qid)` will now not satisfy LSP and would need to be updated?

Reply:
yeah agree this is less than happy, let me see if I can work around the other issue (we currently call on in __call__ with the wrong signature and get away with it)

Reply:
Yeah, I think it wasn't an issue earlier because mypy was not type-checking the method at all, due to no type annotations. But now that's not the case and hence it raises an error. 

We can also decide to make `__call__` a more restrictive type on the base class, so if user overrides `on`, they must also override `__call__` -- but that would also be a breaking change. 

Maybe @maffoo knows some mypy magic to get around the situation without having to deal with breaking changes.

Reply:
> Maybe @maffoo knows some mypy magic to get around the situation without having to deal with breaking changes.

I got nothin', sorry :-)